### PR TITLE
chore: add python linters and formatters to CI

### PR DESCRIPTION
**Pull request by Trinaa**
- Add a job to execute python checks against load test solution in CI
- Add make commands for load tests and lint
- Apply linters/formatters to locustfile.py
- Update README to include mypy
  - The configurations and requirements include mypy, but the README didn't, I erred on the side of caution and included it. 
- (piggyback) optimizing Dockerfile requirements installation and pinning poetry

[SYNC-3827](https://mozilla-hub.atlassian.net/browse/SYNC-3827)

[SYNC-3827]: https://mozilla-hub.atlassian.net/browse/SYNC-3827?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by pjenvey — DISMISSED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -40,6 +40,31 @@ jobs:
...
           command: cargo audit
 
+  load-test-checks:
+    docker:
+      - image: cimg/python:3.11
+    working_directory: "~/autopush-rs/tests/load"
+    steps:
+      - checkout:
+          path: ~/autopush-rs/
+      - run:
```
Comment (pjenvey) [inline .circleci/config.yml:50]: Not critical but these steps could live in the top level Makefile instead

**Comment by jrconlin** [comment]
@Trinaa Looks like the merge of #407 caused a few conflicts. They look fairly simple to resolve. Sorry.

**Comment by Trinaa** [comment]
> @Trinaa Looks like the merge of #407 caused a few conflicts. They look fairly simple to resolve. Sorry.

>> No worries! I agree simple to resolve.

File: .circleci/config.yml (hunk 1 above)
Comment (Trinaa) [reply .circleci/config.yml:50]: I didn't see that Makefile! I will move them!

**Review by jrbenny35 — DISMISSED** [review]
Looks good!

File: .circleci/config.yml (hunk 1 above)
Comment (Trinaa) [reply .circleci/config.yml:50]: @pjenvey Make commands added, can you please check them, MakeFiles are not my specialty.

**Review by jrconlin — COMMENTED** [review]
minor style nit (since I'm the one responsible for it, so :facepalm: 

We should be more consistent with the leading `"""` block and either start comment lines with it (e.g. 
```"""Does this thing...``` 
or 
```""""
Does this thing...
```

We can fix that later, though.

**Review by jrconlin — APPROVED** [review]

**Comment by Trinaa** [comment]
> minor style nit (since I'm the one responsible for it, so 🤦
> 
> We should be more consistent with the leading `"""` block and either start comment lines with it (e.g. `"""Does this thing...` or
> 
> ```
> Does this thing...
> ```
> 
> We can fix that later, though.

Introducing `pydocstyle` to hold us accountable to that is probably the way to go... see [SYNC-3835](https://mozilla-hub.atlassian.net/jira/software/c/projects/SYNC/boards/211?modal=detail&selectedIssue=SYNC-3835)

**Pull request merged**